include/deal.II/lac/solver_gmres.h

```diff
@@ -710,81 +723,319 @@ SolverGMRES<VectorType>::givens_rotation(Vector<double> &h,
 
 
 
-template <class VectorType>
-inline double
-SolverGMRES<VectorType>::iterated_modified_gram_schmidt(
-  const internal::SolverGMRESImplementation::TmpVectors<VectorType>
-    &                                       orthogonal_vectors,
-  const unsigned int                        dim,
-  const unsigned int                        accumulated_iterations,
-  VectorType &                              vv,
-  Vector<double> &                          h,
-  bool &                                    reorthogonalize,
-  const boost::signals2::signal<void(int)> &reorthogonalize_signal)
+namespace internal
 {
-  Assert(dim > 0, ExcInternalError());
-  const unsigned int inner_iteration = dim - 1;
+  namespace SolverGMRESImplementation
+  {
+    template <class VectorType>
+    void
+    dot_and_add(
+      const unsigned int dim,
+      const VectorType & vv,
+      const internal::SolverGMRESImplementation::TmpVectors<VectorType>
+        &             orthogonal_vectors,
+      Vector<double> &h)
+    {
+      for (unsigned int i = 0; i < dim; ++i)
+        h[i] += vv * orthogonal_vectors[i];
+    }
```

Comment:
I am not sure I like the name here. This is a matrix-vector product of a long and skinny matrix `orthogonal_vectors` and a single vector along the *long* side of the matrix, resulting in a *short* vector. In other words, isn't this a `vmult_add` operation in the deal.II language?

Reply:
As a second note, I do not feel strongly in favor for `vmult_add` either, because that would only be in line with our naming conventions if we made it a matrix-vector product of `SolverGMRESImplementation::TmpVectors`. Not sure what is best here, but in general I think that calling it a matrix-vector product (and the other a transposed matrix-vector product) would be the most intuitive naming.

Reply:
What about `Tvmult_add`?